docs/reference/syntax.md

```diff
@@ -0,0 +1,935 @@
+
+```groovy
+file('hello.txt', checkIfExists: true)
+```
+
+The named arguments are collected into a map and provided as the first positional argument to the function. Thus the above function call can be rewritten as:
+
+```groovy
+file([checkIfExists: true], 'hello.txt')
+```
+
+The argument name must be an identifier or string literal.
+
+When the function call is also an [expression statement](#expression-statement) and there is at least one argument, the parentheses can be omitted:
+
+```groovy
+// positional args
+printf 'Hello %s!\n', 'World'
+
+// positional and named args
+file 'hello.txt', checkIfExists: true
+```
+
+If the last argument is a closure, it can be specified outside of the parentheses:
+
+```groovy
+// closure arg with additional args
+[1, 2, 3].inject('result:') { acc, v -> acc + ' ' + v }
+
+// single closure arg
+[1, 2, 3].each() { v -> println v }
+
+// single closure arg without parentheses
+[1, 2, 3].each { v -> println v }
+```
+
+### Constructor call
+
+A constructor call consists of the `new` keyword followed by a *type name* and an argument list enclosed in parentheses:
+
+```groovy
+new java.util.Date()
+```
+
+If the type is implicitly available in the script, the *fully-qualified type name* can be elided to the *simple type name*:
+
+```groovy
+new Date()
+```
+
+Refer to {ref}`stdlib-default-imports` for the set of types which are implicitly available in Nextflow scripts.
+
+### Unary expressions
+
+A unary expression consists of a *unary operator* followed by an expression:
+
+```groovy
+!(2 + 2 == 4)
+```
+
+The following unary operators are available:
+
+- `~`: bitwise NOT
+- `!`: logical NOT
+- `+`: unary plus
+- `-`: unary minus
+
+### Binary expressions
+
+A binary expression consists of a *left expression* and a *right expression* separated by a *binary operator*:
+
+```groovy
+2 + 2
+```
+
+The following binary operators are available:
+
+- `**`: power (i.e. exponentiation)
+- `*`: multiplication
+- `/`: division
+- `%`: remainder (i.e. modulo)
+- `+`: addition
+- `-`: subtraction
+- `<<`: left shift
+- `>>`: right shift
+- `>>>`: unsigned right shift
+- `..`: inclusive range
+- `..<`: right-exclusive range
+- `as`: type cast
+- `instanceof`: type relation
+- `!instanceof`: negated type relation
+- `<`: less than
+- `>`: greater than
+- `<=`: less than or equals
+- `>=`: greater than or equals
+- `in`: membership
+- `!in`: negated membership
+- `==`: equals
+- `!=`: negated equals
+- `<=>`: spaceship (i.e. three-way comparison)
+- `=~`: regex find
+- `==~`: regex match
+- `&`: bitwise AND
+- `^`: bitwise XOR (exclusive or)
+- `|`: bitwise OR
+- `&&`: logical AND
+- `||`: logical OR
+- `?:` elvis (i.e. short ternary)
+
+### Ternary expression
+
+A ternary expression consists of a *test expression*, a *true expression*, and a *false expression*, separated by a question mark and a colon:
+
+```groovy
+x % 2 == 0 ? 'x is even!' : 'x is odd!'
```

Comment:
I've thought with a few of these that it'd be nice to show the example being assigned to a variable.

When it's so "naked" it feels a little too abstract. Personally I'd find this easier to understand:

```nextflow
def is_even = x % 2 == 0 ? 'x is even!' : 'x is odd!'
```

Could also have more verbose explanation of this one:

```nextflow
expression ? value_if_true : value_if_false
```

Reply:
I can add a `println` so that it will at least print something when copied into a script

Reply:
Though in general I don't want to add too much extra stuff especially around these expressions. It's becoming increasingly clear to me that this page is more of a "language specification" rather than a guide, so it can be a little abstract